README-Substrait.md

```diff
@@ -0,0 +1,173 @@
+# Substrait Dialect for MLIR
+
+This project consist of building an input/output dialect in
+[MLIR](https://mlir.llvm.org/) for [Substrait](https://substrait.io/), the
+cross-language serialization format of database query plans (akin to an
+intermediate representation/IR for database queries). The immediate goal is to
+create common infrastructure that can be used to implement consumers, producers,
+optimizers, and transpilers of Substrait; the more transcending goal is to study
```

Comment:
The transcending goal feels to local. Using compilation techniques to optimize database queries?

Reply:
You mean "using MLIR to implement" something isn't broad/ambitious/transcendental enough? How about referring to one of the properties of MLIR then? "using a generic compiler framework (like MLIR)" or "using a progressive lowering approach"?

"Using compilation techniques to optimize database queries" makes me think in different directions: one is the old textbook-style "query compilation," which DB systems do basically since they were invented, and the other one is to use code generation (through LLVM), which some DB systems have been doing for the last 10-15 years.

Maybe "using modern, general-purpose compiler infrastructure to ..."? "Modern" referring to generic, modular, based on progressive lowering, ... as pioneered/popularized by MLIR? Also, I think "compiler infrastructure" is minimally more different from "compilation" and thus associates less with the textbook DB compilers, at least in my ears.